File: cml-compiler/cml-generator/src/test/resources/templates/lang/py/expressions/employment.py
class _Employment:

    _singleton = None

    def __new__(cls) -> '_Employment':
        if cls._singleton is None:
            cls._singleton = super(_Employment, cls).__new__(cls)
        return cls._singleton

    def __init__(self) -> 'None':
        self.__employer = {} # type: Dict[Employee, Organization]
        self.__employees = {} # type: Dict[Organization, List[Employee]]

    def link_many(self, employer: 'Organization', employees: 'List[Employee]') -> 'None':
        for employee in employees: self.link(employer, employee)

    def link(self, organization: 'Organization', employee: 'Employee') -> 'None':
        self.__employer[employee] = organization

        if organization in self.__employees:
            employee_list = self.__employees[organization]
        else:
            employee_list = [employee]
        if not (employee in employee_list):
            employee_list.append(employee)
        self.__employees[organization] = employee_list

    def employer_of(self, employee: 'Employee') -> 'Organization':
        if employee in self.__employer:
            return self.__employer[employee]
        else:
            return None

    def employees_of(self, organization: 'Organization') -> 'List[Employee]':
        if organization in self.__employees:
            employee_list = self.__employees[organization]
        else:
            employee_list = []
        return list(employee_list)

File: cml-modules/cml_base/tests/associations/bidirectional/expected/vehicles_cmlc_java/src/main/java/vehicles/cmlc/Employment.java
package vehicles.cmlc;

import java.util.*;
import java.math.*;
import org.jetbrains.annotations.*;

public class Employment
{
    private static final Employment singleton = new Employment();

    static void init(Class<?> cls)
    {
        if (Employee.class.isAssignableFrom(cls))
        {
            EmployeeImpl.setEmployment(singleton);
        }
        if (Organization.class.isAssignableFrom(cls))
        {
            OrganizationImpl.setEmployment(singleton);
        }
    }

    private final Map<Employee, Organization> employer = new HashMap<>();
    private final Map<Organization, Set<Employee>> employees = new HashMap<>();

    synchronized void linkMany(Organization employer, List<Employee> employees)
    {
        for (Employee employee: employees) link(employer, employee);
    }

    synchronized void link(Organization organization, Employee employee)
    {
        this.employer.put(employee, organization);

        final Set<Employee> employeeList = this.employees.computeIfAbsent(organization, key -> new LinkedHashSet<>());
        if (!employeeList.contains(employee))
        {
            employeeList.add(employee);
        }
    }

    synchronized Optional<Organization> employerOf(Employee employee)
    {
        return Optional.ofNullable(this.employer.get(employee));
    }

    synchronized List<Employee> employeesOf(Organization organization)
    {
        final Set<Employee> employeeList = this.employees.get(organization);

        return (employeeList == null) ? Collections.emptyList() : new ArrayList<>(employeeList);
    }
}